src/main.rs

```diff
@@ -1,3 +1,37 @@
+use std::thread;
+use std::time::Duration;
+
+mod networking;
+mod cache;
+mod model;
+
 fn main() {
-    println!("Hello, world!");
+
+	println!("Hyperledgerいろは2にようこそ！");
+
+    // Setup data structures
+	let block_time_ms = 1000; //TODO: read from config file
+
+	// move block creation to another thread
+    thread::spawn(move || {
+        loop {
+            thread::sleep(Duration::from_millis(block_time_ms));
+            println!("building block");
+        }
+    });
```

Comment:
To my mind it is better to work with **async** functions then threads - easier to use, especially with io-bound operations. Maybe something like:
```
async fn create_blocks(){
    thread::sleep(Duration::from_millis(block_time_ms));
    println!("building block");
}
```
And then in **main**: `let future = create_blocks();`

Reply:
I think it is better to use async functions inside the logic that will create blocks. This is mainly a placeholder, because sleeping here would be pretty stupid in practice if there is other work to do (like verifying transactions).

Reply:
asyncs should be used only for async IO operations
current solution has no reason to spawn a new thread